collector/src/test/java/com/usthe/collector/util/JsonPathParserTest.java

```diff
@@ -1,21 +1,120 @@
+                    "        \"title\": \"Sword of Honour\",\n" +
+                    "        \"price\": 12.99\n" +
+                    "      },\n" +
+                    "      { \"category\": \"fiction\",\n" +
+                    "        \"author\": \"Herman Melville\",\n" +
+                    "        \"title\": \"Moby Dick\",\n" +
+                    "        \"isbn\": \"0-553-21311-3\",\n" +
+                    "        \"price\": 8.99\n" +
+                    "      },\n" +
+                    "      { \"category\": \"fiction\",\n" +
+                    "        \"author\": \"J. R. R. Tolkien\",\n" +
+                    "        \"title\": \"The Lord of the Rings\",\n" +
+                    "        \"isbn\": \"0-395-19395-8\",\n" +
+                    "        \"price\": 22.99\n" +
+                    "      }\n" +
+                    "    ],\n" +
+                    "    \"bicycle\": {\n" +
+                    "      \"color\": \"red\",\n" +
+                    "      \"price\": 19.95\n," +
+                    "      \"gears\": [23, 50]\n," +
+                    "      \"extra\": {\"x\": 0}\n," +
+                    "      \"escape\" : \"Esc\\b\\f\\n\\r\\t\\u002A\",\n" +
+                    "      \"nullValue\": null\n" +
+                    "    }\n" +
+                    "  }\n" +
+                    "}";
     @BeforeEach
     void setUp() {
     }
 
     @Test
     void parseContentWithJsonPath() {
+        // process array
+        List<Object> tom = JsonPathParser.parseContentWithJsonPath(JSON_ARRAY,"$[0].name");
+        assertNotNull(tom);
+        assertEquals("tom",tom.get(0));
+        // get json array map
+        List<Object> entry = JsonPathParser.parseContentWithJsonPath(JSON_ARRAY,"$[1]");
+        assertNotNull(entry);
+        entry.forEach(e -> {
+            assertInstanceOf(Map.class, e);
+            assertEquals("543",((Map)e).get("speed"));
+        });
+        // process object
+        List<Object> author = JsonPathParser.parseContentWithJsonPath(JSON_OBJECT,"$.store.book[0].author");
+        assertNotNull(author);
+        assertEquals("Nigel Rees",author.get(0));
+        // get json object map
+        List<Object> book = JsonPathParser.parseContentWithJsonPath(JSON_OBJECT,"$.store.book[1]");
+        assertNotNull(book);
+        book.forEach(e -> {
+            assertInstanceOf(Map.class, e);
+            assertEquals("Sword of Honour",((Map)e).get("title"));
+        });
+    }
+
+    /**
+     * @throws java.lang.UnsupportedOperationException: Json-smart provider does not support TypeRef! Use a Jackson or Gson based provider
+     * need provid an provider to support TypeRef,like this:
+     *   final Configuration configuration = Configuration.builder()//
+     *         .jsonProvider(new JacksonJsonProvider(Json.mapper()))//
+     *         .mappingProvider(new JacksonMappingProvider(Json.mapper()))//
+     *         .build();
+     */
+//    @Test
```

Comment:
Great! LGTM. Thanks.
Is the `parseContentWithJsonPath2 ` need Test. Please add `@test`